src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double COEFFICIENT = 1.5;
+
+    private T[] array = (T[]) new Object[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexBounds(index < 0);
+        ensureCapacity();
+        if (index == size) {
+            add(value);
+        } else {
+            T t = array[index];
+            checkIndexBounds(t == null);
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexBounds(isCorrectIndex(index));
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexBounds(isCorrectIndex(index));
```

Comment:
The same issue as in line 43 is present here. The isCorrectIndex method is returning incorrect results. The check should ensure the index is within the bounds of the array size.